src/hotspot/share/utilities/concurrentHashTable.hpp

```diff
@@ -406,10 +407,11 @@ class ConcurrentHashTable : public CHeapObj<F> {
                       size_t log2size_limit = DEFAULT_MAX_SIZE_LOG2,
                       size_t grow_hint = DEFAULT_GROW_HINT,
                       bool enable_statistics = DEFAULT_ENABLE_STATISTICS,
+                      Mutex::Rank rank = Mutex::nosafepoint-2,
```

Comment:
This is the default rank.  It seems reasonable to not stop for safepoints with the resizing lock (since it's the lock taken while in a safepoint - do_safepoint_scan).  This was the default before it was lowered to service-1.

Reply:
Ah. Sorry for missing it. I was thrown off by the explicit constructor below and the different order of arguments.

Reply:
It took me while to find it from Thomas's original patch too.  Thanks!